packages/strapi-admin/admin/src/utils/ComponentApi.js

```diff
@@ -0,0 +1,42 @@
+import { cloneDeep } from 'lodash';
+import invariant from 'invariant';
+
+class ComponentApi {
+  components = {};
+
+  getComponent = name => {
+    invariant(name, 'A name must be provided');
+
+    return cloneDeep(this.components[name]) || null;
+  };
+
+  getComponents = () => {
+    const components = cloneDeep(this.components);
+
+    return Object.keys(components).reduce((acc, current) => {
+      acc[current] = components[current].Component;
+
+      return acc;
+    }, {});
+  };
+
+  registerComponent = component => {
+    const { name, Component } = component;
+
+    invariant(Component, 'A Component must be provided');
+    invariant(name, 'A name must be provided');
+    invariant(this.components[name] === undefined, 'A similar field already exists');
+
+    this.components[name] = { Component };
+  };
```

Comment:
Is there a way to register multiple components?

Reply:
Like in the field api noe